book/src/configuration.md

```diff
@@ -123,6 +123,7 @@ The following statusline elements can be configured:
 | `auto-signature-help` | Enable automatic popup of signature help (parameter hints)  | `true`  |
 | `display-inlay-hints` | Display inlay hints[^2]                                     | `false` |
 | `display-signature-help-docs` | Display docs under signature help popup             | `true`  |
+| `display-inline-diagnostics`  | Display diagnostics under their starting line       | `true`  |
```

Comment:
yeah the docs need to be updated that's mostly why it's still a draft. These are not boolean options anymore since you can filter the minimum diagnostic level (and set a separate level for the visual line containing the cursor). For now that means that you won't be able to use `:toggle` with it but once #4411 is merged that will be possible again

Reply:
any hints how to configure it?

```
[editor.lsp]
inline-diagnostics.cursor-line = [] // works
```
```
// doesn't work
inline-diagnostics.other-lines = ["error"]
inline-diagnostics.other-lines = "error"
inline-diagnostics.other-lines = { "error" }
inline-diagnostics.other-lines = { "at-least": "error" }
```

```
// no idea
[keys.normal]
"A-D" = [":set lsp.inline-diagnostics.cursor-line []", ":set lsp.inline-diagnostics.other-lines []"]
"A-E" = [":set lsp.inline-diagnostics.cursor-line [\"hint\", \"info\", \"warning\", \"error\"]", ":set lsp.inline-diagnostics.other-lines [\"warning\", \"error\"]"]
```

Reply:
I fixed the config, I have only tested by hardcoding the config so far and noticed that the derived deserialized wasn't great (expected `{"severity" : "error"}`). Now the following works:
* `[]` (disabled)
* `["warning", "error"]` (only warnings and errors)
* "info" (info or above, so equivalent to `["info", "warning", "error"]`

Arrays don't seem to work with `:set` right now but that's kind of a separate issue. There is quite a bit of config options that don't work correctly with `:set` probably something that should be addressed separately (see #6392 for example)

Reply:
ok, for others with hyperactive lsp - this hack works for me:

```
[keys.normal]
"A-D" = ":set lsp.inline-diagnostics.min-diagnostic-width 1024"
"A-E" = ":set lsp.inline-diagnostics.min-diagnostic-width 40"
```

Reply:
you can get this to work with quoting `:set lsp.inline-diagnostics "[ \"error\" ]"`` for example